src/main/java/dev/streamx/cli/publish/PublishCommand.java

```diff
@@ -0,0 +1,81 @@
+package dev.streamx.cli.publish;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import dev.streamx.cli.ingestionclient.IngestionClientContext;
+import dev.streamx.cli.ingestionclient.IngestionClientException;
+import dev.streamx.cli.ingestionclient.StreamxClientProvider;
+import dev.streamx.cli.publish.payload.PayloadResolver;
+import dev.streamx.clients.ingestion.StreamxClient;
+import dev.streamx.clients.ingestion.exceptions.StreamxClientException;
+import jakarta.inject.Inject;
+import picocli.CommandLine.Command;
+import picocli.CommandLine.Help.Visibility;
+import picocli.CommandLine.Model.CommandSpec;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.ParameterException;
+import picocli.CommandLine.Parameters;
+import picocli.CommandLine.Spec;
+
+@Command(name = "publish", mixinStandardHelpOptions = true)
+public class PublishCommand implements Runnable {
+
+  @Parameters(index = "0", description = "Channel that message will be published to")
+  String channel;
+
+  @Parameters(index = "1", description = "Message key")
+  String key;
+
+  @Option(names = "--ingestion-url",
```

Comment:
as an improvement, we could read it from .env file or properties

Reply:
There are multiple posibilities. I was thinking about `namespace` concepts like in kubectl. And I think they may be conflicting approaches.

Moreover in current scope we usualy follow conventions, that ingestion url is localhost:8080 (in this task, we focus on publishing to local mesh).
Enabling other environments should be done in following tasks.